src/Aquifer.Jobs/Subscribers/TranslationSubscriber.cs

```diff
@@ -0,0 +1,383 @@
+                message.StartedByUserId,
+                projectResourceContentIds,
+                Queues.GetPoisonQueueName(Queues.TranslateProjectResources),
+                queueMessage.MessageText),
+            ct);
+
+        _logger.LogInformation(
+            "Kicked off {FunctionName} for Project ID {ProjectId}.",
+            nameof(OrchestrateProjectResourcesTranslation),
+            message.ProjectId);
+    }
+
+    /// <summary>
+    /// Durable orchestration function.
+    /// This function is single-threaded and thus should not do any DB operations or any substantial processing.
+    /// It should only orchestrate activities.
+    /// </summary>
+    [Function(nameof(OrchestrateProjectResourcesTranslation))]
+    public async Task OrchestrateProjectResourcesTranslation(
+        [OrchestrationTrigger] TaskOrchestrationContext context,
+        OrchestrateProjectResourcesTranslationDto dto)
+    {
+        try
+        {
+            var translateResourceTasks = dto.ProjectResourceContentIds
+                .Select(resourceContentId => context.CallActivityAsync<int>(
+                    nameof(TranslateProjectResourceActivity),
+                    new TranslateProjectResourceActivityDto(resourceContentId, dto.StartedByUserId),
+                    s_durableFunctionTaskOptions))
+                .ToList();
+
+            await Task.WhenAll(translateResourceTasks);
+
+            var translatedResourceContentIds = translateResourceTasks
+                .Select(trt => trt.Result)
+                .ToList();
+
+            await context.CallActivityAsync(
+                nameof(UpdateProjectPostTranslationActivity),
+                new UpdateProjectPostTranslationActivityDto(dto.ProjectId, dto.StartedByUserId, translatedResourceContentIds),
+                s_durableFunctionTaskOptions);
+        }
+        catch (Exception orchestrationException)
+        {
+            // All the Activities orchestrated here have a retry policy.  If things are still failing after multiple retries
+            // then something is very wrong. If this happens then swallow errors here to allow successful orchestration function
+            // completion but publish a poison queue item for manual retry later.
+            var ct = CancellationToken.None;
+
+            try
+            {
+                _logger.LogError(
+                    orchestrationException,
+                    "Error translating resource content for Project ID {ProjectId}. A poison message will be published to \"{PoisonQueueName}\" to enable manual retry. Manual dev intervention is required.",
+                    dto.ProjectId,
+                    dto.PoisonQueueName);
+
+                var translateProjectResourcesPoisonQueueClient = await _queueClientFactory.GetQueueClientAsync(
+                    dto.PoisonQueueName,
+                    ct);
+                await translateProjectResourcesPoisonQueueClient.SendMessageAsync(dto.OriginalQueueMessageText, ct);
```

Comment:
I like this pattern of publishing a message back to the poison queue of the original subscriber function if the orchestration fails.  The original subscriber has no way of tracking when the orchestration completes; it simply triggers the process and completes successfully and then all of the translations run in the background.

Reply:
Good question.  For a single queue subscriber that fails, yes, the poison queue logic happens automatically.

However, the translate project resources queue subscriber can only kick off the durable functions orchestration.  It doesn't wait for the orchestration to finish (due to how durable functions work).  Thus, so long as it successfully kicks off the orchestration the subscriber will succeed and won't publish to the poison queue, even if the orchestration eventually fails.

Therefore, I manually published to the poison queue if the orchestration fails so we can easily retry the project translation.